src/test/stub/len_vec.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+extern crate libc;
+
+use std::mem;
+use std::ptr::{write, read};
+
+// Abstraction which tracks only the length and the last element of the vector.
+pub struct Vec<T> {
+    len: usize,
+    last: Option<*mut T>,
+}
+
+impl<T> Vec<T> {
+    fn drain(&mut self) {
+        self.len = 0;
+        self.last = None;
+    }
+
+    fn update_last(&mut self, elem: T) {
+        if self.len == 0 {
+            let elem_size = mem::size_of::<T>();
+            let ptr = unsafe { libc::malloc(elem_size) as *mut T };
+            self.last = Some(ptr);
+        }
+
+        unsafe {
+            write(self.last.unwrap().offset(0), elem);
+        }
+    }
+
+    pub fn new() -> Vec<T> {
+        Vec {
+            len: 0,
+            last: None
+        }
+    }
+
+    pub fn with_capacity(_: usize) -> Self {
+        Vec {
+            len: 0,
+            last: None
+        }
+    }
+
+    pub fn push(&mut self, elem: T) {
+        if self.len == 0 {
+            let elem_size = mem::size_of::<T>();
+            let ptr = unsafe { libc::malloc(elem_size) as *mut T };
+            self.last = Some(ptr);
+        }
+        self.len += 1;
+        unsafe {
+            write(self.last.unwrap().offset(0), elem);
+        }
+    }
+
+    pub fn pop(&mut self) -> Option<T> {
+        if self.len == 0 {
+            None
+        } else {
+            self.len -= 1;
+            unsafe {
+                Some(read(self.last.unwrap().offset(0)))
+            }
+        }
+    }
+
+    pub fn append(&mut self, other: &mut Vec<T>) {
+        if other.len() != 0 {
+            self.update_last(other.pop().unwrap());
+            self.len += other.len();
+        }
+    }
+
+    pub fn insert(&mut self, index: usize, elem: T) {
+        assert!(index <= self.len);
+
+        if index == self.len {
```

Comment:
Should this be index out of bounds?
I see, it does a shift.  Maybe comment that?